GitWorkflow/GitWorkflow/Player.cpp

```diff
@@ -21,29 +21,43 @@ void Player::collide(const Platform& platform)
 
 void Player::update(const float dt)
 {
-	if (std::roundf(m_velocity.y) == 0.0f && (sf::Keyboard::isKeyPressed(sf::Keyboard::LShift) || sf::Keyboard::isKeyPressed(sf::Keyboard::RShift))) m_speed = m_baseSpeed * 1.5f;
+	if (m_collidedLastFrame && 
+		(sf::Keyboard::isKeyPressed(sf::Keyboard::LShift) || sf::Keyboard::isKeyPressed(sf::Keyboard::RShift))) 
+		m_speed = m_baseSpeed * m_sprintModifier;
 	else m_speed = m_baseSpeed;
 
-	if (sf::Keyboard::isKeyPressed(sf::Keyboard::A)) { if (std::abs(m_velocity.x) < m_speed * (std::roundf(m_velocity.y) == 0.0f ? 1.0f : 0.5f)) m_velocity.x -= m_acceleration * dt * (std::roundf(m_velocity.y) == 0.0f ? 1.0f : 0.5f); }
+	const float control = m_collidedLastFrame ? 1.0f : m_airControl;
+
+	if (sf::Keyboard::isKeyPressed(sf::Keyboard::A)) 
+	{ 
+		if (std::abs(m_velocity.x) < m_speed * control) 
+			m_velocity.x -= m_acceleration * control * dt;
+	}
 	else if (m_velocity.x < 0.0f) m_velocity.x += m_acceleration * dt;
 
-	if (sf::Keyboard::isKeyPressed(sf::Keyboard::D)) { if (std::abs(m_velocity.x) < m_speed * (std::roundf(m_velocity.y) == 0.0f ? 1.0f : 0.5f)) m_velocity.x += m_acceleration * dt * (std::roundf(m_velocity.y) == 0.0f ? 1.0f : 0.5f); }
-	else if (m_velocity.x > 0.0f) m_velocity.x -= m_acceleration * dt;
+	if (sf::Keyboard::isKeyPressed(sf::Keyboard::D)) 
+	{
+		if (std::abs(m_velocity.x) < m_speed * control)
+			m_velocity.x += m_acceleration * dt * control;
+	}
+	else if (m_velocity.x > 0.0f) m_velocity.x -= m_acceleration * control * dt;
```

Comment:
These two else if statements will cause the player to alternate between a positive and negative (albeit small) x-velocity.
After both keys are checked, resolve velocity in one place instead of one, and if it's less than a specified threshold it should be set to zero.

Reply:
This should be fixed in most recent branch version.